site/src/docs/foundation/guidelines/server-side-rendering.mdx

```diff
@@ -0,0 +1,171 @@
+---
+slug: "/foundation/guidelines/server-side-rendering"
+title: "Server-side rendering"
+navTitle: 'Server-side rendering'
+---
+import { Accordion, Notification } from 'hds-react';
+import LeadParagraph from '../../../components/LeadParagraph';
+
+# Server-side rendering
+
+<LeadParagraph>
+  The server-side rendering support of the Helsinki Design System makes the user's landing experience to pages smooth without
+  flashes of unstyled content.
+</LeadParagraph>
+
+## What is server-side rendering?
+
+Server-side rendering (SSR) is the process where the rendering of html pages is done on the server-side. The fully rendered html
+document is then sent to the browser to be viewed. The rendering of the html can happen at build time (Static site generation
+or pre-rendering), or during an HTTP request. The alternative to SSR is client-side rendering (CSR), where rendering of the
+html happens in the browser.
+
+## How does HDS support server-side rendering?
+
+For HDS components to work with server-side rendering, in server you need to inject the critical CSS styles
+of the HDS components that are being used into the initially rendered html. For hds-core, you must include the styles from
+provided CSS files yourself. For hds-react, HDS provides multiple options, that we will cover next.
+
+<Notification label="Critical CSS" className="siteNotification">
+  Critical CSS is the CSS applied to above-the-fold elements. It is the CSS responsible for content immediately
+  visible when a user opens your website. Pages can be long with scrollable content, that is not visible immediately.
+  There's no need to include the styles of elements that are not visible. There is one exception though, and that is
+  if you have anchor links on the page. When user opens website from an url that includes an anchor link, browser
+  automatically scrolls the page so that the link will be visible.
+</Notification>
+
+### Option 1: getCriticalHdsRules tool to get critical styles (recommended)
+
+HDS React components use CSS-in-JS in such a way, that the styles are by default injected to head tag in browser. If you
```

Comment:
~~to~~ **into the** head in **the** browser

Reply:
- [ ] Fix articles